cibuildwheel/pyodide.py

```diff
@@ -0,0 +1,385 @@
+            if compatible_wheel:
+                log.step_end()
+                print(
+                    f"\nFound previously built wheel {compatible_wheel.name}, that's compatible with {config.identifier}. Skipping build step..."
+                )
+                built_wheel = compatible_wheel
+            else:
+                if build_options.before_build:
+                    log.step("Running before_build...")
+                    before_build_prepared = prepare_command(
+                        build_options.before_build, project=".", package=build_options.package_dir
+                    )
+                    shell(before_build_prepared, env=env)
+
+                log.step("Building wheel...")
+
+                extra_flags = split_config_settings(build_options.config_settings, "build")
+                extra_flags += build_frontend.args
+
+                if not 0 <= build_options.build_verbosity < 2:
+                    msg = f"build_verbosity {build_options.build_verbosity} is not supported for build frontend. Ignoring."
+                    log.warning(msg)
+
+                build_env = env.copy()
+                if build_options.dependency_constraints:
+                    build_env["PIP_CONSTRAINT"] = str(constraints_path)
+                build_env["VIRTUALENV_PIP"] = get_pip_version(env)
+                call(
+                    "pyodide",
+                    "build",
+                    build_options.package_dir,
+                    f"--outdir={built_wheel_dir}",
+                    *extra_flags,
+                    env=build_env,
+                )
+                built_wheel = next(built_wheel_dir.glob("*.whl"))
+
+                if built_wheel.name.endswith("none-any.whl"):
+                    raise NonPlatformWheelError()
+
+                if build_options.repair_command:
+                    log.step("Repairing wheel...")
+
+                    repair_command_prepared = prepare_command(
+                        build_options.repair_command,
+                        wheel=built_wheel,
+                        dest_dir=repaired_wheel_dir,
+                    )
+                    shell(repair_command_prepared, env=env)
+                else:
+                    shutil.move(str(built_wheel), repaired_wheel_dir)
+
+                repaired_wheel = next(repaired_wheel_dir.glob("*.whl"))
+
+                if repaired_wheel.name in {wheel.name for wheel in built_wheels}:
+                    raise AlreadyBuiltWheelError(repaired_wheel.name)
+
+            if build_options.test_command and build_options.test_selector(config.identifier):
+                log.step("Testing wheel...")
+
+                venv_dir = identifier_tmp_dir / "venv-test"
+                # set up a virtual environment to install and test from, to make sure
+                # there are no dependencies that were pulled in at build time.
+
+                # --no-download??
+                call("pyodide", "venv", venv_dir, env=env)
```

Comment:
One of the things that did not work out of the box on a local build test was that I had a broken `node` in my PATH. This led to tests failure when creating the test virtual environment.
That means that the node version from emscripten is not used here. We should probably use that node version to limit differences in build environments.

Reply:
We would not want to do this. Generally the node from Emscripten is quite old. It currently includes node 16, but the earliest node that Pyodide fully supports is node v18 (in node v16 http requests won't work correctly unless you apply a manual fix that we've dropped from Pyodide). Starting from node v20 we can make `loop.run_until_complete()` work correctly which will allow us to test packages that use asyncio much more naturally.

Reply:
ok so using the node version from Emscripten is out. Should we add what's needed to install a pinned version of node (e.g. latest LTS version) to prevent environment changes then ?

Reply:
Sounds reasonable.

Reply:
Ok , I'll look into adding a commit for this then.

Reply:
Thanks for the help @mayeut!